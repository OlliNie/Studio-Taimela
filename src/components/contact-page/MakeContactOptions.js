import React from 'react';
import styles from './Contact.css';

export default function MakeContactOptions({ contactOptions }) {
  const contact = contactOptions.map((contactOption, i) => (
    <li className={styles.li} key={i}>
      <p className={styles.logo}>{contactOption.icon}</p>
      <p>{contactOption.text}</p>
    </li>
  ));
  return contact;
}
