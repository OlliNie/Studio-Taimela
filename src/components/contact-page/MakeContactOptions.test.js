import { shallow } from 'enzyme';
import React from 'react';
import MakeContactOptions from './MakeContactOptions';

describe('ContactOption component', () => {
  it('renders a list of contact options from an array', () => {
    const contactOptionsArray = [
      {
        icon: 'fake icon',
        text: 'Virpi@studiotaimela.com',
      },
      {
        icon: 'fake icon2',
        text: (
          <>
            <span>ZÃ¼rich</span>
            <span>Switzerland</span>
          </>
        ),
      },
      {
        icon: `fake icon3`,
        text: (
          <>
            <span>+41 77 979 9734</span>
            <span> +358 40 410 6428</span>
          </>
        ),
      },
    ];

    const wrapper = shallow(
      <MakeContactOptions contactOptions={contactOptionsArray} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
