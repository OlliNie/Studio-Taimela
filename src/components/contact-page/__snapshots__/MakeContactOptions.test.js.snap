// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactOption component renders a list of contact options from an array 1`] = `
<ul
  className="ul"
>
  <li
    className="li"
    key="0"
  >
    <p
      className="logo"
    >
      <FaEnvelopeOpenText
        size={32}
      />
    </p>
    <p>
      Virpi@studiotaimela.com
    </p>
  </li>
  <li
    className="li"
    key="1"
  >
    <p
      className="logo"
    >
      <FaMapMarked
        size={32}
      />
    </p>
    <p>
      <span>
        ZÃ¼rich
      </span>
      <span>
        Switzerland
      </span>
    </p>
  </li>
  <li
    className="li"
    key="2"
  >
    <p
      className="logo"
    >
      <FaPhone
        size={32}
      />
    </p>
    <p>
      <span>
        +41 77 979 9734
      </span>
      <span>
         +358 40 410 6428
      </span>
    </p>
  </li>
</ul>
`;
